src/app_service/service/community/service.nim

```diff
@@ -250,6 +254,21 @@ QtObject:
         stopped: receivedData.stopped
       ))
 
+    self.events.on(SignalType.DownloadingHistoryArchivesStarted.event) do(e: Args):
+      var receivedData = HistoryArchivesSignal(e)
+      if receivedData.communityId notin self.historyArchiveDownloadTaskCommunityIds:
+        self.historyArchiveDownloadTaskCommunityIds.incl(receivedData.communityId)
+        self.events.emit(SIGNAL_COMMUNITY_HISTORY_ARCHIVES_DOWNLOAD_STARTED, CommunityIdArgs(communityId: receivedData.communityId))
+
+    self.events.on(SignalType.DownloadingHistoryArchivesFinished.event) do(e: Args):
+      var receivedData = HistoryArchivesSignal(e)
+      if receivedData.communityId in self.historyArchiveDownloadTaskCommunityIds:
+        self.historyArchiveDownloadTaskCommunityIds.excl(receivedData.communityId)
+
+      if self.historyArchiveDownloadTaskCommunityIds.len == 0:
```

Comment:
I'm not sure about this condition. Shouldn't we emit the event under line 266 above?

Reply:
It doesn't really matter. Both works

Reply:
Wouldn't it cause issues if we have two communities downloading? When one finishes, we still want to inform that one of them is done? I know we don't handle that currently (only one banner for all communities).
Should we just add a TODO here to change it later? We could just remove the community passing for now, because it implies that the signal really is per community.

Reply:
> Wouldn't it cause issues if we have two communities downloading? When one finishes, we still want to inform that one of them is done? I know we don't handle that currently (only one banner for all communities).

So right now we really don't we're only interested if all of them are done.
But I see where you're coming from. This seem unintuitive. I'll update this and add a comment as well for clarity.